subprojects/gradle/kotlin-dsl-support/src/main/kotlin/com/avito/kotlin/dsl/ProjectExtensions.kt

```diff
@@ -5,56 +5,58 @@ import org.gradle.api.file.RegularFileProperty
 import java.io.File
 import kotlin.reflect.KProperty
 
-fun Project.getOptionalStringProperty(name: String): String? {
-    return if (hasProperty(name)) {
-        property(name)?.toString()
-    } else {
-        null
-    }
-}
+//TODO these methods should return Property<X>
 
-fun Project.getOptionalStringProperty(name: String, default: String): String {
-    return if (hasProperty(name)) {
-        property(name)?.toString() ?: default
+/**
+ * @param nullIfBlank there may be cases when it's ok to pass empty string to override
+ * todo true by default, false to not break anything that rely on previous behavior
+ */
+fun Project.getOptionalStringProperty(name: String, nullIfBlank: Boolean = false): String? =
+    if (hasProperty(name)) {
+        val string = property(name)?.toString()
+        if (nullIfBlank && string.isNullOrBlank()) null else string
     } else {
-        default
+        null
     }
-}
 
-fun Project.getMandatoryStringProperty(name: String): String =
-    getOptionalStringProperty(name) ?: throw RuntimeException("Parameter: $name is required (must be not empty)")
+fun Project.getOptionalStringProperty(name: String, default: String, defaultIfBlank: Boolean = true): String =
+    getOptionalStringProperty(name, nullIfBlank = defaultIfBlank) ?: default
 
-fun Project.getOptionalIntProperty(name: String): Int? = try {
-    getOptionalStringProperty(name)?.toInt()
-} catch (e: NumberFormatException) {
-    null
-}
+/**
+ * @param allowBlank todo false by default
+ */
+fun Project.getMandatoryStringProperty(name: String, allowBlank: Boolean = true): String =
+    getOptionalStringProperty(name, nullIfBlank = !allowBlank)
+        ?: throw RuntimeException("Parameter: $name is required (must be not empty)")
```

Comment:
It's better to distinquish two different errors:

- "Parameter: $name is missing but required" <--- hasProperty == false
- "Parameter: $name is blank but required" <--- blank value

It'd easier to debug errors.